File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30ObserverPatternJava/src/main/java/ec/edu/espe/model/Investor.java

package ec.edu.espe.model;


/**
 *
 * @author MSI-PULSE
 */
public class Investor implements Observer {
    private String name;

    public Investor(String name) {
        this.name = name;
    }

    @Override
    public void update(String stockSymbol, double stockPrice) {
        System.out.println("Investor " + name + " notified. " + 
                            "Stock: " + stockSymbol + " Price: " + stockPrice);
    }

    public String getName() {
        return name;
    }
}

File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30ObserverPatternJavaScript/model/Investor.js
class Investor extends Observer {
    constructor(name) {
        super();
        this.name = name;
    }

    update(stockSymbol, stockPrice) {
        console.log(`Investor ${this.name} notified. Stock: ${stockSymbol} Price: ${stockPrice}`);
    }

    getName() {
        return this.name;
    }
}
